### PR TITLE
[SHIMGVW] Fix image flickering when rendering animations

### PR DESCRIPTION
**Pull request by binarymaster**
## Purpose

When shell image viewer renders animations, there is flickering between frames. This PR fixes it.

JIRA issue: None

## Proposed changes

- Do `InvalidateRect` without clearing background
- Use `NULL_BRUSH` to make `Rectangle` transparent (avoid extra clear)
- Do not fill entire window with `FillRect`, instead fill regions around the image

Cc @katahiromz and @yagoulas 

Note for Giannis:

> I also tried to make usage of `WndClass.hbrBackground` but not appreciated it (it flickers the entire window when resizing, even toolbar :open_mouth: )

### CODE REVIEW

File: dll/win32/shimgvw/shimgvw.c [hunk 1]
```diff
@@ -633,6 +651,7 @@ ImageView_DrawImage(HWND hwnd)
             GdipSetSmoothingMode(graphics, SmoothingModeHighQuality);
         }
 
+        SelectObject(hdc, GetStockObject(NULL_BRUSH));
```
Comment (ThFabba) [inline dll/win32/shimgvw/shimgvw.c]: Should this be undone after? Not sure what the previous brush is but it seems like it could be leaked.

Comment (katahiromz) [reply dll/win32/shimgvw/shimgvw.c]:
```c
HGDIOBJ hbrOld;
...
hbrOld = SelectObject(hdc, GetStockObject(NULL_BRUSH));
...
SelectObject(hdc, hbrOld);
```

File: dll/win32/shimgvw/shimgvw.c [hunk 2]
```diff
@@ -587,8 +587,10 @@ ImageView_DrawImage(HWND hwnd)
     UINT ImageWidth, ImageHeight;
     INT ZoomedWidth, ZoomedHeight, x, y;
     PAINTSTRUCT ps;
-    RECT rect;
+    RECT rect, clean;
```
Comment (katahiromz) [inline dll/win32/shimgvw/shimgvw.c]: s/clean/margin/

File: dll/win32/shimgvw/shimgvw.c [hunk 3]
```diff
@@ -609,14 +611,31 @@ ImageView_DrawImage(HWND hwnd)
...
     {
-        FillRect(hdc, &rect, (HBRUSH)GetStockObject(WHITE_BRUSH));
-
         ZoomedWidth = (ImageWidth * ZoomPercents) / 100;
         ZoomedHeight = (ImageHeight * ZoomPercents) / 100;
 
         x = (rect.right - ZoomedWidth) / 2;
         y = (rect.bottom - ZoomedHeight) / 2;
 
+        white = GetStockObject(WHITE_BRUSH);
```
Comment (HBelusca) [inline dll/win32/shimgvw/shimgvw.c]: Why white? Shouldn't it be of the default (background) color of windows?

Comment (binarymaster) [reply dll/win32/shimgvw/shimgvw.c]: - It looks good with this color :)
- It was done with this color before
- Also this color is used by shimgvw in Windows

Comment (HBelusca) [reply dll/win32/shimgvw/shimgvw.c]: Thanks for the precisions :)

**Pull request merged**